File: backend/internal/models/review.go
package models

type Review struct {
	ID          int    `json:"id"`
	ReviewerID  int    `json:"reviewer_id"`
	RevieweeID  int    `json:"reviewee_id"`
	AuctionID   int    `json:"auction_id"`
	RatingStars int    `json:"rating_stars"`
	Comment     string `json:"comment"`
	CreatedAt   string `json:"created_at"`

	ReviewerName  string `json:"reviewer_name,omitempty"`
	ReviewerImage string `json:"reviewer_image,omitempty"`
}

type CreateReviewRequest struct {
	RevieweeID  int    `json:"reviewee_id" validate:"required,gt=0"`
	AuctionID   int    `json:"auction_id" validate:"required,gt=0"`
	RatingStars int    `json:"rating_stars" validate:"required,min=1,max=5"`
	Comment     string `json:"comment" validate:"required,max=500"`
}

File: yutu_frontend/lib/models/review.dart
class Review {
  final int id;
  final int reviewerId;
  final int revieweeId;
  final int auctionId;
  final int ratingStars;
  final String comment;
  final String createdAt;
  final String reviewerName;
  final String reviewerImage;

  Review({
    required this.id,
    required this.reviewerId,
    required this.revieweeId,
    required this.auctionId,
    required this.ratingStars,
    required this.comment,
    required this.createdAt,
    required this.reviewerName,
    required this.reviewerImage,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'] ?? 0,
      reviewerId: json['reviewer_id'] ?? 0,
      revieweeId: json['reviewee_id'] ?? 0,
      auctionId: json['auction_id'] ?? 0,
      ratingStars: json['rating_stars'] ?? 0,
      comment: json['comment'] ?? '',
      createdAt: json['created_at'] ?? '',
      reviewerName: json['reviewer_name'] ?? 'Anonymous',
      reviewerImage: json['reviewer_image'] ?? '',
    );
  }
}